enterprise/backend/src/metabase_enterprise/sso/integrations/jwt.clj

```diff
@@ -137,9 +137,10 @@
             (str return-to-param redirect))))))
 
 (defmethod sso.i/sso-get :jwt
-  [{{:keys [jwt redirect]} :params, :as request}]
+  [{{:keys [jwt access_token redirect]} :params, :as request}]
   (premium-features/assert-has-feature :sso-jwt (tru "JWT-based authentication"))
-  (let [jwt-data (when jwt (session-data jwt request))
+  (let [jwt (or jwt access_token)
```

Comment:
The access_token should be an opaque token and is not supposed to be decoded. I'm not sure the full steps of the keycloak repro, but a standards compliant version would accept a parameter called `id_token` that has a jwt with identity claims.